clinica/iotools/utils/pipeline_handling.py

```diff
@@ -58,10 +60,114 @@ def t1_freesurfer_pipeline(caps_dir, df, freesurfer_atlas_selection=None, **kwar
     Returns:
          final_df: a DataFrame containing the information of the bids and the pipeline
     """
+
     from clinica.iotools.converters.adni_to_bids.adni_utils import (
         replace_sequence_chars,
     )
     from clinica.utils.stream import cprint
+    from pathlib import Path
+
+    # Ensures that df is correctly indexed
+    if "participant_id" in df.columns.values:
+        df.set_index(["participant_id", "session_id"], inplace=True, drop=True)
```

Comment:
```suggestion
        df.set_index(["participant_id", "session_id"], inplace=True, verify_integrity=True)
```

Reply:
And what to do if the dataframe is actually not sound (ValueError due to failed integrity check)?

Reply:
Updated to cprint an error message and exit

Reply:
Same comment as above, we probably want to unconditionnally run set_index with the expected column names (no if case) to reject unsound input dataframes.

Reply:
Fixed